src/network/packets.h

```diff
@@ -78,4 +91,145 @@ enum class PacketErrorCode : uint8_t {
 	OTHER = 5,
 };
 
+#pragma pack(push, 1)
+
+template <typename T>
+T swapEndianness(T value) {
+	auto* bytes = reinterpret_cast<uint8_t*>(&value);
+	std::reverse(bytes, bytes + sizeof(T));
+	return value;
+}
```

Comment:
it does not convert here? returning value and not byte?
Not sure why the server seems to get values that way but seems odd.

Reply:
It does, bytes holds a pointer to the memory location value is at. Therefore reversing those bytes reverses the bytes in the value too.